analyzers/src/SonarAnalyzer.CSharp.Styling/Rules/LambdaParameterName.cs

```diff
@@ -29,7 +29,7 @@ public sealed class LambdaParameterName : StylingAnalyzer
         context.RegisterNodeAction(c =>
             {
                 var parameter = ((SimpleLambdaExpressionSyntax)c.Node).Parameter;
-                if (parameter.Identifier.ValueText != "x"
+                if (parameter.Identifier.ValueText is not ("x" or "_")
```

Comment:
What about leaving a comment here that in C# 13, it will be a real discard and the `"_"` will not be needed anymore?

Reply:
Even if parsed as a discard, it is still an IdenifierToken:
[sharplab.io](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgUGYACdQgYUIG9dCbiiUAWQgWQAoBKS62nlDNADwBLAHYxRwAHyEwhALyFWAfWiElnOdIoBfANzceNZYVUTCQjYozQM7fTh7bc2oA)

So I think it will still be needed.

Reply:
Do you know if there's a branch available with that feature? I didn't find any in a quick search.

It might not be IdentifierToken, but some kind of discard designator, like here:
https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgUGYACdQgYUIG9dCbiiUAWQgWQAoBKS62ngNzABOhVgH1ohAB6cAvIQAqHANzcaAXxWENABwEBLfsACmw3QDsYhM8E4KZAPkKxDAMzABXADbBlOVUA===

Reply:
I'm not aware of a branch for it. The closest I could find on SharpLab is "Semi Auto-properties," but that one hasn't been updated since Oct 22.

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLTIgWwJaqnGAe1QAcFCTkAaGEKZGKgExAGoAfAAQCYBGAWABQnAMwACHmIDCYgN5CxiieOwA7GGIAK5EnLEBzCBoC8APjFhsEADZMA3GIwwHAXwVL3i0RIAsYgLIAFACUcp5KXrzcADxqMOZQAMYw2ISqYsZiAPoZ5rIuduFugi5AA===

The `DiscardDesignation` is an interesting find. The significant difference here is that the distinguishing happens on the node level, while for parameters, it would likely need to occur on the token level.

There are cases like this already with, e.g., `IdentifierNameSyntax` where an Identifier property can be either an `IdentiferToken` or a `GlobalKeyword`.
There can be `IdentifierToken` or the `ArgListKeyword` for parameters. Maybe we see an `UnderScore` here as well in the future.
```xml
    <Node Name="ParameterSyntax" Base="BaseParameterSyntax">
        <Kind Name="Parameter"/>
        <! -- ... -->
        <Field Name="Identifier" Type="SyntaxToken">
            <PropertyComment>
                <summary>Gets the identifier.</summary>
            </PropertyComment>
            <Kind Name="IdentifierToken"/>
            <Kind Name="ArgListKeyword"/>
        </Field>
        <! -- ... -->
    </Node>
```